dubbo-remoting/dubbo-remoting-http12/src/main/java/org/apache/dubbo/remoting/http12/netty4/h2/NettyHttp2ProtocolSelectorHandler.java

```diff
@@ -68,8 +69,9 @@
             h2StreamChannel = new Http2WriteQueueChannel(h2StreamChannel, writeQueue);
         }
         ChannelPipeline pipeline = ctx.pipeline();
-        pipeline.addLast(
-                new NettyHttp2FrameHandler(h2StreamChannel, factory.newInstance(h2StreamChannel, url, frameworkModel)));
+        Http2TransportListener http2TransportListener = factory.newInstance(h2StreamChannel, url, frameworkModel);
+        ctx.channel().closeFuture().addListener(future -> http2TransportListener.onStreamClosed());
```

Comment:
**Suggestion**: Consider verifying the success of the channel closure operation within the listener to handle potential failures. <br> **Code Suggestion**: 
 ```
 +ctx.channel().closeFuture().addListener(future -> {
 +    if (future.isSuccess()) {
 +        http2TransportListener.onStreamClosed();
 +    } else {
 +        // Handle closure failure
 +    }
 +});
 ```